source/core/src/main/com/csse3200/game/components/player/PlayerActions.java

```diff
@@ -30,15 +28,17 @@ public class PlayerActions extends Component {
   private PhysicsComponent physicsComponent;
   private Vector2 walkDirection = Vector2.Zero.cpy();
   private boolean moving = false;
-  EventService eventService = ServiceLocator.getEventService();
+  private final EventService eventService = ServiceLocator.getEventService();
```

Comment:
The EventService has been removed from main since it violates system design by introducing an unnecessary global variable. You will need to use the players event handling system instead.